packages/medusa-js/src/request.ts

```diff
@@ -183,7 +183,7 @@ class Client {
   async request(
     method: RequestMethod,
     path: string,
-    payload: object = {},
+    payload: object | null = null,
```

Comment:
As per @adrien2p's suggestion lets do this:

```suggestion
    payload: Record<string, unknown> | null = null,
```